vpr/src/place/initial_placement.cpp

```diff
@@ -314,6 +321,69 @@ static t_physical_tile_type_ptr pick_placement_type(t_logical_block_type_ptr log
     return nullptr;
 }
 
+vtr::vector<ClusterBlockId, t_block_score> assign_block_scores() {
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& place_ctx = g_vpr_ctx.placement();
+
+    auto blocks = cluster_ctx.clb_nlist.blocks();
+    auto pl_macros = place_ctx.pl_macros;
+
+    t_block_score score;
+
+    vtr::vector<ClusterBlockId, t_block_score> block_scores;
+
+    for (unsigned int i = 0; i < blocks.size(); i++) {
+        block_scores.push_back(score);
+    }
+
+    //go through all blocks and store floorplan constraints and num equivalent tiles
+    for (auto blk_id : blocks) {
+        if (is_cluster_constrained(blk_id)) {
+            block_scores[blk_id].num_floorplan_constraints = 1;
+        }
+        auto logical_block = cluster_ctx.clb_nlist.block_type(blk_id);
```

Comment:
want to make num_equivalent tiles proportionate to region size

Reply:
may want to just store total score instead of each individual thing like constraints, num_equivalent tiles, etc.